homeassistant/components/zoneminder/binary_sensor.py

```diff
@@ -1,26 +1,40 @@
 """Support for ZoneMinder binary sensors."""
+from typing import Callable, List, Optional
+
+from zoneminder.zm import ZoneMinder
+
 from homeassistant.components.binary_sensor import BinarySensorEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import Entity
 
-from . import DOMAIN as ZONEMINDER_DOMAIN
+from .common import get_client_from_data
 
 
-async def async_setup_platform(hass, config, add_entities, discovery_info=None):
-    """Set up the ZoneMinder binary sensor platform."""
-    sensors = []
-    for host_name, zm_client in hass.data[ZONEMINDER_DOMAIN].items():
-        sensors.append(ZMAvailabilitySensor(host_name, zm_client))
-    add_entities(sensors)
-    return True
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: Callable[[List[Entity], Optional[bool]], None],
+) -> None:
+    """Set up the sensor config entry."""
+    zm_client = get_client_from_data(hass, config_entry.unique_id)
+    async_add_entities([ZMAvailabilitySensor(zm_client, config_entry)])
 
 
 class ZMAvailabilitySensor(BinarySensorEntity):
     """Representation of the availability of ZoneMinder as a binary sensor."""
 
-    def __init__(self, host_name, client):
+    def __init__(self, client: ZoneMinder, config_entry: ConfigEntry):
         """Initialize availability sensor."""
         self._state = None
-        self._name = host_name
+        self._name = config_entry.unique_id
```

Comment:
Isn't this a breaking change because the name is now changing?

Reply:
No. The unique_id of the config is the host name. https://github.com/home-assistant/core/pull/37060/files#diff-50e12b59b701cc09cf1dd70145ff9dcbR96